core/src/mast/serialization/mod.rs

```diff
@@ -4,21 +4,29 @@
 //! - MAGIC
 //! - VERSION
 //!
-//! (lengths)
+//! (sections metadata)
 //! - decorators length (`usize`)
+//! - decorator data section offset (`usize`) (NOT YET IMPLEMENTED)
 //! - nodes length (`usize`)
 //!
 //! (procedure roots)
 //! - procedure roots (`Vec<MastNodeId>`)
 //!
-//! (raw data)
+//! (basic block data section)
+//! - basic block data
+//!
+//! (node info structs)
+//! - MAST node infos (`Vec<MastNodeInfo>`)
+//!
+//! (raw decorator data)
```

Comment:
Nit: Maybe we can call all the sections "section", e.g. "node info section", etc.. This makes it clearer that the offsets in sections metadata point to those sections.

E.g.:

```
//! The serialization format of MastForest is as follows:
//!
//! (Metadata)
//! - MAGIC
//! - VERSION
//!
//! (sections metadata)
//! - nodes length (`usize`)
//! - decorator data section offset (`usize`)
//! - decorators length (`usize`)
//!
//! (procedure roots section)
//! - procedure roots (`Vec<MastNodeId>`)
//!
//! (basic block data section)
//! - basic block data
//!
//! (node info section)
//! - MAST node infos (`Vec<MastNodeInfo>`)
//!
//! (decorator data section)
//! - Decorator data
//! - String table
//!
//! (decorator info section)
//! - decorator infos (`Vec<DecoratorInfo>`)
//!
//! (basic block decorator lists section)
//! - basic block decorator lists (`Vec<(MastNodeId, Vec<(usize, DecoratorId)>)>`)
//! 
//! (before enter and after exit decorators section)
//! - before enter decorators (`Vec<(MastNodeId, Vec<DecoratorId>)>`)
//! - after exit decorators (`Vec<(MastNodeId, Vec<DecoratorId>)>`)
```

Reply:
@PhilippGackstatter makes sense. I will remove `decorator data section offset (`usize`)` per request from @plafer since it will be implemented later.

Is there an GitHub issue for truncating decorators already?

Reply:
Oh with https://github.com/0xPolygonMiden/miden-vm/pull/1531#discussion_r1848615075 I initially assumed the decorator data section was implemented already but the comment was not updated - my bad.

I don't believe there's an issue for it, so yes let's just create an issue and maybe update the comment to:

```rs
//! - decorator data section offset (`usize`) (not implemented, see issue #X)
```